### PR TITLE
Oracle 11 XE is now fired up in a docker image - looks like we get back ...

### PR DESCRIPTION
**Pull request by DeepDiver1975**
...Oracle Testing on pull requests

@PVince81 @icewind1991 @MorrisJobke

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
On Jenkins: https://ci.owncloud.org/job/server-master-linux-oracle-in-docker/

**Comment by PVince81** [comment]
```
16:17:03 Kill the docker 
16:17:03 
16:17:03 Usage: docker rm [OPTIONS] CONTAINER [CONTAINER...]
16:17:03 
16:17:03 Remove one or more containers
16:17:03 
16:17:03   -f, --force=false      Force the removal of a running container (uses SIGKILL)
16:17:03   -l, --link=false       Remove the specified link and not the underlying container
16:17:03   -v, --volumes=false    Remove the volumes associated with the container
16:17:04 Build step 'Execute shell' marked build as failure
16:17:04 Recording test results
16:17:05 Finished: FAILURE
```

**Comment by PVince81** [comment]
Nice one, now I see what you meant when you said "need to create a database first"

File: autotest.sh [hunk 1]
```diff
@@ -72,6 +73,8 @@ if [ -f config/config.php ] && [ ! -f config/config-autotest-backup.php ]; then
 fi
 
 function cleanup_config {
+    echo "Kill the docker $DOCKER_CONTAINER_ID"
```
Comment (MorrisJobke) [inline autotest.sh]: Intendation and check if this is even set ;)

**Comment by DeepDiver1975** [comment]
@PVince81 @MorrisJobke addressed

File: autotest.sh [hunk 2]
```diff
@@ -72,6 +73,11 @@ if [ -f config/config.php ] && [ ! -f config/config-autotest-backup.php ]; then
 fi
 
 function cleanup_config {
+    if [ ! -z "$DOCKER_CONTAINER_ID" ]; then
```
Comment (nickvergessen) [inline autotest.sh]: spaces instead of tabs?

File: autotest.sh [hunk 3]
```diff
@@ -117,13 +123,22 @@ function execute_tests {
 		dropdb -U $DATABASEUSER $DATABASENAME || true
 	fi
 	if [ "$1" == "oci" ] ; then
+		echo "Fire up the oracle docker"
+		DOCKER_CONTAINER_ID=`docker run -d wnameless/oracle-xe-11g`
+		DATABASEHOST=`docker inspect $DOCKER_CONTAINER_ID | grep IPAddress | cut -d '"' -f 4`
+
+		echo "Waiting 60 seconds for Oracle initialization ... "
+		sleep 60
```
Comment (nickvergessen) [inline autotest.sh]: cant we jsut try the "drop database" in a loop, instead of waiting 60seconds?

**Comment by PVince81** [comment]
Would it be possible to use sqlplus from the docker instance instead of expecting a local one ?

```
± % ./autotest.sh oci | tee test.log 2>&1 ; less test.log                                                                                  !10451
Using database oc_autotest
Setup environment for oci testing ...
Fire up the oracle docker
Waiting 60 seconds for Oracle initialization ... 
drop the database
sqlplus -s -l system/oracle@//172.17.0.2:1521/xe
./autotest.sh: line 136: sqlplus: command not found
Kill the docker 6cb030ff890b83cc25fca6c2fc49c5a0cdacee7aa0f6a9cde1962efb5eeb7b46
6cb030ff890b83cc25fca6c2fc49c5a0cdacee7aa0f6a9cde1962efb5eeb7b46
```

**Comment by DeepDiver1975** [comment]
> Would it be possible to use sqlplus from the docker instance instead of expecting a local one ?

well - you require the php module for oracle as well - in order to compile this you need the oracle instant client which ships sqlplus.

What we could do is to build our own docker image on top of wnameless/oracle-xe-11g which also creates the database.

**Comment by DeepDiver1975** [comment]
> Would it be possible to use sqlplus from the docker instance instead of expecting a local one ?

@PVince81 ok - here we go - no sqlplus required

**Comment by DeepDiver1975** [comment]
green ball - https://ci.owncloud.org/job/server-master-linux-oracle-in-docker/9/

**Comment by MorrisJobke** [comment]
:+1:

**Comment by icewind1991** [comment]
:+1:

**Pull request merged**